### DISCUSSION TITLE
pre/post setLocale hooks

### DISCUSSION THREAD

**Discussion by nkoehring** [ivanhofer/typesafe-i18n#544] [DX improvements]
It would be very useful to be able to globally register hooks that run before or after setting a locale. These could be used to do things like setting `<html lang="$newLocale">` or `axios.defaults.headers.common['Accept-Language'] = locale`.

**Comment by ivanhofer** [comment]
Hi @nkoehring,
since you as a developer are writing the code that changes locale, you already have full control over it. You just need to add the lines of code before/after you call `setLocale`. In my opinion, there is no need to add a hook for this functionality. Or do I miss something?

**Reply by nkoehring** [reply]
Generally I agree. When I got that idea, I thought about applications, where different places would potentially set change locale, but to avoid code duplication, one could simply write a wrapper.

**Reply by ivanhofer** [reply]
Yes, abstracting it into a reusable function. That's the way I would do it.
